tests/sqlgateway/mock/mock_client.py

```diff
@@ -0,0 +1,62 @@
+import os
+from typing import List
+
+from flink.sqlgateway.operation import SqlGatewayOperation
+from flink.sqlgateway.session import SqlGatewaySession
+from flink.sqlgateway.config import SqlGatewayConfig
+from tests.sqlgateway.mock.gw_router import GwRouter
+import requests
+import json
+
+
+class MockFlinkSqlGatewayClient:
+    def __init__(self):
+        self.router = None
+        self.session = None
+
+    def create_session(self, host: str, port: int, session_name: str) -> SqlGatewaySession:
+        host_port = f"http://{host}:{port}"
+        test_config = {
+            "host_port": host_port,
+            "schemas": [
+                {"catalog": "default_catalog", "database": "default_database", "tables": [], "views": []}
+            ],
+            "current_catalog": "default_catalog",
+            "current_database": "default_database",
+        }
+        self.router = GwRouter(test_config)
+        self.router.start()
+        # create session
+        r = requests.post(f"{host_port}/v1/sessions", json.dumps({"sessionName": f"{session_name}"}))
+        session_handle = r.json()['sessionHandle']
+        session = SqlGatewaySession(SqlGatewayConfig(host, port, session_name), session_handle)
+        self.session = session
+        return session
+
+    def execute_statement(self, sql: str) -> SqlGatewayOperation:
+        if self.session.session_handle is None:
+            raise Exception(
+                f"Session '{self.session.config.session_name}' is not created. Call create() method first"
+            )
+        host_port = f"http://{self.session.config.host}:{self.session.config.port}"
+        session_handle = self.session.session_handle
+        data = {"statement": sql}
+        r = requests.post(f"{host_port}/v1/sessions/{session_handle}/statements", json.dumps(data))
+        operation_handle = r.json()['operationHandle']
+        return SqlGatewayOperation(session=self.session, operation_handle=operation_handle)
+
+    def clear_statements(self):
+        return self.router.clear_statements()
+
+    def all_statements(self) -> List[str]:
+        return self.router.all_statements()
+
+    @staticmethod
+    def setup(host: str = None, port: str | int = None, session_name: str = None):
```

Comment:
the union `|` operator is as far as I am aware available from 3.10. It has unfortunately still used by very few people https://w3techs.com/technologies/history_details/pl-python/3 
in setup.py we are stating this library is using Python 3.7 or newer. Can we get rid of this for now? Maybe lets standardize port as `int` type.

Reply:
ok